common/transport/amqp/src/receiver_link.ts

```diff
@@ -0,0 +1,208 @@
+import * as machina from 'machina';
+import * as amqp10 from 'amqp10';
+import * as dbg from 'debug';
+import { EventEmitter } from 'events';
+import { Message, errors, results } from 'azure-iot-common';
+import { AmqpMessage } from './amqp_message';
+import { AmqpLink } from './amqp_link_interface';
+
+const debug = dbg('ReceiverLink');
+
+/*Codes_SRS_NODE_AMQP_RECEIVER_LINK_16_002: [** The `ReceiverLink` class shall inherit from `EventEmitter`.]*/
+/*Codes_SRS_NODE_AMQP_RECEIVER_LINK_16_003: [** The `ReceiverLink` class shall implement the `AmqpLink` interface.]*/
+export class ReceiverLink  extends EventEmitter implements AmqpLink {
```

Comment:
When the link is detached it gets an error that we then forward in the _onEnter handler of the detached state. at that point an error event is emitted. so whoever is using the link needs to subscribe to the error event to know that it is detached. if the error event is not subscribed to the program will throw (which I think is the correct behavior).


---
In reply to: [127619355](https://github.com/Azure/azure-iot-sdk-node/pull/55#discussion_r127619355) [](ancestors = 127619355)